### ISSUE TITLE
Enhanced navigation suggestions/feedback

### ISSUE THREAD

**Issue by reetp** [RocketChat/Rocket.Chat#33561]
Testing Enhanced navigation on 6.12.2

General consensus is the new nav is mainly a retrograde step.

Contextual bar resize in threads is good, but could be allowed to be wider, or even full width?

The new 'Enhanced' nav feels disjointed and in 2024 we do not need everything bunched up into tiny spaces like we would with mobile - we have enough screen estate not to worry about that. It is simplification gone way to far.

We have wide screens so moving the Settings and Profile to the far right is totally incongruous. They are lost.

![Screenshot_2024-10-14_11-23-39](https://github.com/user-attachments/assets/91b339d1-6df5-4de4-ae5a-e0ee6999949d)

We prefer all main settings grouped together in one place, not split like this.

We do not like the Omnichannel being lumped in with other settings. The old way is MUCH clearer with omnichannel chats and settings in their own clearly defined section. We'd prefer more delineation than less. We have plenty of screen space....

With settings are compressed the only separation between icons is a small vertical bar which tells you nothing, and the hover text doesn't help either.

![RocketLayout](https://github.com/user-attachments/assets/cccb0089-17db-4df6-83cf-b2af8bba43b5)

We don't suppose this will get listened too and we will have to suffer it being enforced but it is what it is. Unpopular.

**Comment by ivans-netto** [comment]
Hi @reetp 

Redesigning Rocket.Chat's navigation is indeed challenging, and we’ve kept it under feature preview precisely to gather insights like these and ensure we implement meaningful changes. The iterative approach allows us to fine-tune the experience as we progress toward the larger vision. I believe some of our next steps can solve some of the cases you noticed during your tests

I’d love to set up a chat to discuss more. 

I'll be reaching out to you on open.rocket.chat.

**Comment by reetp** [comment]
As this now appears to be close to release I'm adding my open thoughts here. Looking forward to seeing what the public make of this when it is released.

> Enhanced navigation
> Reduce noise and increase focus using the new global header and sidebar filters 

Allegedly.

--------------------

The second column is absolutely awful, unnecessary distraction, a total eyesore, slows your brain down, and made worse because it now seems that it cannot be disabled. Absolutely shocking.

Utter over engineered junk.

So exactly how do I quickly find a channel or DM now? Before it was just the quick search eye glass above.

<img width="761" height="514" alt="Image" src="https://github.com/user-attachments/assets/77eec4ec-79ae-4186-92ca-395c9a5ced65" />

Now? Please don't tell me it's the stupid search bar at the top? That takes way longer to find anything (and I just finally understood what it does - it is SO unintuitive I though it did a channel or global search. Fail.)

This was a neater idea if you really want it.

https://github.com/RocketChat/feature-requests/issues/961

Or why not a single column where you can drag and drop in groups as you want them? A smarter way to do it.

--------------------

Splitting the settings and main icons is still just utter mindless vandalism. On wide screens it takes me ages to go tracking back and forth trying to find the setting I want. it feels like it has been done to annoy me rather than help me.

Note: 640 x 480 went out with Ark. I don't need icons all tightened to nothing. My screens are 2560 wide. They can cope with lots of icons, but I don't need lots more travel.

Keeping Omnichannel in it's own group, and on the left where most of the action is, was far more logical too. Clear delineation. But now it is lost and hard to find.

Reducing the channel importance was just ridiculous - it is where most of us work most of the time. Again, I don't need a tiny thin channel bar. I am not trying to save a pixel.

And the (ridiculous) search bar just takes up channel bar room. Over engineering to solve an issue that never existed.

--------------------

In summary, if this is all so cool, where's the marketing saying 'try this new fancy thing'?

The almost total embarrassed silence about this speaks volumes.

Seems a Windows Metro moment to me.

**Comment by reetp** [comment]
Going to @dougfabris here as well.

This issue was primarily relating to the top nav bar, rather than the side/second column. 

Please see my first post here. 

As far as I'm aware your PR:

https://github.com/RocketChat/Rocket.Chat/pull/37285 

relates to this the issue rather than the side bar/second column issue here:

https://github.com/RocketChat/Rocket.Chat/issues/36727 

Note I consider this issue with the top nav bar much more important, especially on big monitors. 

However I have heard no serious commentary on my points about over simplification, splitting the controls rather than keeling them in a group, the defined omnichannel section getting lost, and search being more dominant than the channel which you are actually working in.

I'd appreciate some feedback on why you think the changes are good and what benefits they bring.

**Comment by dougfabris** [comment]
Hello @reetp I'm carefully detailing all the feedbacks and issues regarding specifically the enhanced navigation (top nav bar) and I will be opening a discussion with our product lead and our CTO in order to get their response before moving ahead with the changes planned to the next major release `8.0.0`. As soon as I get their response, either if we will change the approach or why we're going to move ahead the way it was planned, I will let you know. 

Thank you for your time, help and patience!

**Comment by milton-rucks** [comment]
Hey @reetp 

Thanks for the feedback. Here are more details on some topics:

**1 - Issue regarding the new breakpoint**
The breakpoint was changed because keeping both sidebars open at 1024px or 768px takes up too much space and competes with the area dedicated to the chat and contextual bar. 
Given we are not promoting secondary sidebar to production anymore, we will revert the breakpoint to 768px.

**2 - The distance of the navbar items on the right side of the screen**
There is not really a market standard for it. This one is more of a personal preference.
I’m taking as an example Jira, which is an application that I use frequently.
<img width="1916" height="217" alt="Image" src="https://github.com/user-attachments/assets/701c95af-ad5f-4dae-845f-ec341d12ad4e" />

**3 - Omnichannel section**
We hear your feedback about differentiating omnichannel features from team collaboration. Although having a separate session in the sidebar makes the difference more evident, we decided to adopt a one header for every room type. We moved to a single, consistent header component. The bar gives a light-weight “sub-group” inside that unified header without forking layouts.
Like any other navigation change, it requires user adaptation, but we consider it will be quickly assimilated by frequent omnichannel users.

**4 - Secondary sidebar**
We reached the conclusion that this feature isn’t mature enough for general availability. We will keep this item as a feature preview and potentially explore more options over the next year.

**Comment by reetp** [comment]
Thanks Milton. It has only taken a year......

1. Why not just let the user decide what size they want?
Who says it takes up too much space? You? Someone else? Clients?
Some want less, others want more - eg me. I have dual 2560*1440 screens. I am not pushed for screen space. Isn't it up to the user how wide they want their panes?
I'm sure you could make it a % percentage of width as well rather than limiting it with px? Isn't that the point of CSS?
Currently on my 6.10.3 the thread pane can be expanded to about 50% of the channel width - around 1024px.  Sufficiebnt though I sometime wish it was wider in threads.
Reducing it to 768px is just silly and yet another retrograde step. *I* certainly don't want it. I want it to be as big as *I* want.

2. Nav bar
> This one is more of a personal preference.

Yup.

> _I’m taking as an example Jira,
I don't really care what Jira does as I don't use it. So it is irrelevant. This is about Rocket, which I do use. 

As I pointed out above first with wide screens it makes the distance ridiculous. It splits all your controls. If you are moving them right, then move them ALL, not half an half so you keep forgetting which side of the screen you need to be on and end up flicking miles left and right until you lose your patience and switch is all off again.

You have body swerved round the issues of jamming the search bar centre stage and reducing the size of the channel information so it is barely distinguable from the chanhel chat.

Then you jam the channel control in a mess under the other buttons and you end up hopelessly lost becuase there is nothing to distinguish anything. Again, with bigger screens we do not tobe minimal. I don't need a single tiny row with tiny icons and tiny letters I can't read with my older eyes. It isn't more accessbile, it is less.

<img width="2557" height="82" alt="Image" src="https://github.com/user-attachments/assets/b8b46fb6-392f-4b8c-824e-91a7c78dd917" />

"Spot the channel and wonder what the icons might do"


3. Omnichannel
 Joy. Well I can tell you as I did before it is a dogs breakfat now. It's slapped in with general settings and profile? Ridiculous.

Yopu single line is just totally unnecessary minimisatin in an era when screens get bigger. We do not use 640x480 any more. At least not here.

It was better like it was in it's own distinguishable section.

I don't dispute the old layout could have done with some tweak,s but IMHO this poorly designed a dogs breakfast.
 
Remember - you have been doing this for months. You have got used to it. 

I and my users come at it cold and I can tell you it is absolutely jarring and they constantly hit off again - it is a massive barrier to productivity.

It'll be interesting to see what happens when you drop this unannounced on most unuspecting users.

4. Second channel - probably the only sensible decision so far. It is horrible. Allowing people to drag andf drop channels into groups with fold down menus (as some one mentioned in a NFR) would be much more flexible.

**Comment by reetp** [comment]
Hey @milton-rucks 

Can you confirm some results of your external user feedback before this is foisted on an unwitting user base?

Love to know how popular this feature is and how many times it is enabled/disabled/remained in use etc.

As product manager overseeing this you have done this basic research, haven't you?

**Comment by reetp** [comment]
"Thorin sits down and starts singing about gold"

https://media4.giphy.com/media/v1.Y2lkPTZjMDliOTUycjAzaDZzc29nbW5qenB3Y2NlMGIxd2F6YjBqeGlwbTN2aGUyMWN2cSZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/1Zbeweu52ZaQE/giphy.gif

Truly unbelievable.

**Comment by ivans-netto** [comment]
Hi @reetp 

Ivan, head of design here. 

This project is currently on hold due to other priorities, and it won’t move out of Feature Preview until it goes through a deeper review. What we released so far is the foundation we started to build, a structural basis to enable the important improvements we want to make. It doesn’t mean those improvements are already included; priorities simply pulled us in other directions before we could complete the work.

The only part of this effort that is being released out of Feature Preview is the new Global Header, which is moving forward independently because it’s stable and aligns with our current roadmap.

While in Feature Preview, we gathered a large amount of feedback, including yours, and all of it has been reviewed and consolidated to inform the next phase. I can’t share the full plans yet, but I can assure you that many of the points you raised are directly influencing our direction.

This project has a lot of potential, and both the community and our customers are being heard. The next steps are now much clearer: what needs to be removed, what should be revamped, what should be reviewed, and what needs to be added.

As mentioned, Feature Preview exists exactly for this purpose, to gather feedback and validate direction, especially for projects that deeply affect how people interact with the Rocket.Chat platform.

Thanks again for taking the time to provide your input. It’s helping shape what comes next.

**Comment by reetp** [comment]
Thanks Ivan.

> The only part of this effort that is being released out of Feature Preview is the new Global Header, which is moving forward independently because it’s stable and aligns with our current roadmap.

You clearly haven't bothered reading what I said right at the start here, or at least totally ignored it.

The 'Global Header' is my primary cause for concern.

"Stable" it may be, but it is still an absolutely awful piece of junk design as described above, and which you have studiously avoided. It's clearly a done deal and you are going to change it no matter what anyone says.

Douglas Adams nailed this in HHGTTG decades ago.

> 'What d you mean you don't like it? It has been in Feature Preview for ages - didn't you look?'

> “But the (UI) plans were on display…”
> “On display? I eventually had to go down to the cellar to find them.”
> “That’s the display department.”
> “With a flashlight.”
> “Ah, well, the lights had probably gone.”
> “So had the stairs.”
> “But look, you found the notice, didn’t you?”
> “Yes,” said Arthur, “yes I did. It was on display in the bottom of a locked filing cabinet stuck in a disused lavatory with a sign on the door saying ‘Beware of the Leopard'” 

He we are just now. 7.12.2 on a Macbook - not even my normal large monitors.

Still trying to cram a quart into a pint pot for absolutely no visible reason at all, and in a time when screens get ever bigger. Buttons fished randomly out of a drawer with no clue what they do. Trying to cram topic links into the same line as the channel name - we use them a lot on some channels and making it one line instead of two for absolutely no gain is a complete pain. Omnichannel hiding in plain site alongside settings with no delineation?

My original comments still stand. It is horrendous. A barrier to productivity. It is your own Windows Metro.
 
<img width="1423" height="52" alt="Image" src="https://github.com/user-attachments/assets/0b4c7d4e-1073-4589-bb6e-d2a6bdfad419" />


Despite the patronising, no one has really answered anything technically right from the start. It is ignored because the reality is no one in the team has an answer. But you are just going to do it anyway. 

I know that saying it is a "fractal of bad design" suggests designers have wasted a year of their life designing it, but then we've all done that at some point in our lives. It happens.

I have been criticising this for over a year, taking time with diagrams and explanations as to why it is wrong, and had not one rational documented answer or proof to tell me why it is an improvement over the current bar, and exactly how popular it is.

Not a single one.

Just avoidance, patronising comments, criticism of me, smoke and mirrors. Anything to avoid actually giving an answer.

<sigh>

That sinking feeling when you know people have fingers in their ears and singing about gold.

I give up. You can lead a horse to water.......

Good luck, I'm sure someone will love it. At least I can vote with my feet, which happily I shall be doing soon.

Closing this because this is clearly an exercise in futility.
